### PR TITLE
[Ready for review] Ignore _foreach_values in merge artifacts

### PR DESCRIPTION
**Pull request by darinyu**
`_foreach_values` is now stored as artifacts for each step but this is not intended for users to consume. In merge artifacts, however, we will try to merge them by default, which could fail if the foreach are dynamic. This PR will make merge artifacts skip the foreach value (at join step). This foreach value will have no meaning after joining anyways, so there should be no impact to the entire flow.

### CODE REVIEW

File: metaflow/flowspec.py [hunk 1]
```diff
@@ -419,6 +419,8 @@ def merge_artifacts(
             This exception is thrown in case an artifact specified in `include` cannot
             be found.
         """
+        INTERNAL_ARTIFACTS_SET = set(["_foreach_values"])
```
Comment (romain-intel) [inline metaflow/flowspec.py:422]: nit: move this out of the function so it doesn't get recreated every time.

**Review by romain-intel — APPROVED** [review]
Minor nit but LGTM.

File: metaflow/flowspec.py (hunk 1 above)
Comment (darinyu) [reply metaflow/flowspec.py]: updated

**Review by romain-intel — APPROVED** [review]
LGTM

**Pull request merged**